lib/arrays.md

```diff
@@ -5,44 +5,60 @@ instructor_notes: Feel free to re-organize the headings (or add/remove headings)
 
 # What is one way, using Ruby, to retrieve the 6th element in an Array? How about the 8th element? What happens if you try to retrieve the value of the _9999th_ element (or some element that doesn't exist in the array)?
 
-Replace me with your response...
+You can use integers to address locations in an Array. So, you could put the index number 6 or 8 in brackets after the Array name to retrieve the 6th or 8th elements (always remembering that counting begins at 0.
+If you tried to retrieve the value of a position in an Array that does not exist, the program would return 'nil', meaning nothing, because there is nothing there.
 
 # The previous question asks about finding, for example, the 6th element in an Array. Is it possible to find the **-6th** (Notice the negative symbol!) element in an Array? What does that even mean?
 
-Replace me with your response...
+Yes. Negative index values count from the end of the Array, so the '-6th' value in an Array would be the sixth value from the end.
 
 # How would you perform an operation on every element inside an Array?
 
-Replace me with your response...
+You could use the method Each to apply an operation to every object an Array points to. 
+If you have an Array called 'NamesofStudents', which includes the name of every student in the upcoming Omaha Code School Cohort, you could use .each to have the computer say hi to each and every student.
+You would use the Each method and a variable that Each can use to point to the objects in an Array, in this case let's say 'name', surrounded by vertical bars.
+Then you can enter what you want the computer to do (in this case let's say write out "Hi, Name!", where Name is actually the individual names of the students) to each object in the Array.
 
 # How do you add elements to an Array?
 
-Replace me with your response...
+You can use the 'Push' method to add an object to the end of an Array. Using the previous example of names of students in the OCS cohort, let's say John Stamos joined late and needs to be added.
+I could put NameofStudents.push 'John Stamos' to add his name to the end of the list.
 
 # Given the Array `["Laura", "Fiona", "Tori"]`, how would you replace `"Fiona"` with `"Florence"` so that you end up with `["Laura", "Florence", "Tori"]`?
 
-Replace me with your response...
+If I knew I wanted to repalce Fiona with Florence, I would search through my Array to find the object "Fiona" using the Each method.
+To do this, I could ask my program to check "Fiona" against each object of an Array to find out where it matches.
+Then, I could write an 'if','else' statement so for each object my program returned that matches "Fiona", it would replace that object with "Florence".
+Alternatively, since we know the position of the element we are trying to modify, we can use Ruby's 'delete_at' and 'insert' methods to remove Fiona from position 1 and add Florence.
```

Comment:
Does that mean that any new assignments I make for specific element in an Array will override all previous ones? So, if the element that I want in position [1] changes to "Fran", can I simply put the_array[1] = "Fran" somewhere later and it will replace "Florence" with "Fran" without getting confused about whether position [1] is Florence or Fran? If so, does subsequent code **always** supersede previous code or is that unique to this circumstance?